src/alert.cpp

```diff
@@ -1548,7 +1548,7 @@ namespace {
 		};
 
 		int const idx = static_cast<int>(op);
-		if (idx < 0 || idx >= int(sizeof(names) / sizeof(names[0])))
```

Comment:
I don't the warning you see here is correct. The cast of ``op`` to an ``int`` can produce a negative value, as far as I can see.

Reply:
enum class operation_t : std::uint8_t

Reply:
I see. However, I like that the current logic here does not *depend* on the underlying storage type of the enum class. I worry about the commit where the 256th operation enum is added, and this type has to be widened, it would be virtually impossible to find all places that depend on it being no more than 8 bits.

In addition, I would expect the compiler to optimise-away this test, so this change is kind of only to please a warning.

I suppose this change would make sense combined with:

``static_assert(std::numeric_limits<std::underlying_type<operation_t>::value>::min() >= 0
&& std::numeric_limits<std::underlying_type<operation_t>::value>::max() <= std::numeric_limits<int>::max(), "an check for negative values is necessary");``.

But it's not obvious to me that that would be an improvement to the code. As it is now, a reader does not need to worry about any of this.